### ISSUE TITLE
Remote works on master version but not on 2.1.3

### ISSUE THREAD

**Issue by JiDai** [guillaumepotier/Parsley.js#1007]
I wrote two pen with a basic example, only with one difference, the version a the parsley remote lib.
2.1.3 : http://codepen.io/JiDai/pen/rOpPGQ
2.2.2rc2 : http://codepen.io/JiDai/pen/jbZbgN
Why no ajax call is fired on 2.1.3 ?
Thanks!

**Comment by marcandre** [comment]
:bomb: Looks like an embarrassing bug in 2.1.3. Not quite sure when that was introduced...
Sorry for this.

**Comment by jplaza** [comment]
So what's the recommended thing to do? Why was this issue closed?

**Comment by marcandre** [comment]
The recommendation is to upgrade to 2.2.0-rc4 :smile:
Let me know if that doesn't work for you

**Comment by jplaza** [comment]
Remote works now, but the form is not submitting at all :cry:

**Comment by jplaza** [comment]
Had to use this ugly hack to be able to use Parsley:

```js

$("form").parsley(window.ParsleyConfig)
  .on('form:submit', function (formInstance) {
    if (formInstance.isValid()) {
      formInstance.destroy();
      $("form").submit();
    }
});

```

**Comment by marcandre** [comment]
The form is not submitting, and there are no validation errors? Do you have a working example? I tried from the above codepen but the ajax request always comes back with a 302 code...?

**Comment by jplaza** [comment]
The validation works as expected, ajax calls are triggered for remote validation, but when everything is ok the form won't submit, then I added the code I posted above to make it work.

This is my form:

```html
<form method="POST" action="/cuentas/registro/negocios/enviar">
    <input type='hidden' name='csrfmiddlewaretoken' value='ajz6O7mD3Hs5OTDMD8qINwNOI2crp7lc' />
    <hr>
    <div class="row">
      <div class="col-md-4 col-sm-3">
        <div class="clearfix form-side-help">
          <div class="form-side-help-icon activate">
          </div>
          <h2>Tu cuenta</h2>
        </div>
        <p>Con estos datos vas a acceder a tu cuenta Dátil. Escribe una dirección de correo electrónico que utilices.</p>
      </div>
      <div class="col-md-7 col-md-offset-1 col-sm-10" style="padding-top: 20px;">
        <!--label for="id_email">Dirección de correo electrónico</label-->
        <div class="row">
          <div class="col-sm-6">
            <div class="form-group ">
  <label for="id_username">Elige un nombre de usuario</label>
  <input class="form-control" id="id_username" maxlength="30" name="username" placeholder="Ejm: jperez" type="text" />
  <div class="help-block">
    
  </div>
</div>
          </div>
          <div class="col-sm-6">
            <div class="form-group ">
  <label for="id_email">Dirección de correo electrónico</label>
  <input class="form-control" id="id_email" name="email" placeholder="joseperez@empresa.com" type="text" />
  <div class="help-block">
    
  </div>
</div>
          </div>
        </div>
        <div class="row">
          <div class="col-sm-6">
            <div class="form-group ">
  <label for="id_password">Contraseña</label>
  <input class="form-control" id="id_password" maxlength="40" name="password" placeholder="Escribe tu contraseña" type="password" />
  <div class="help-block">
    
  </div>
</div>
          </div>
        </div>
      </div>
    </div>
    <hr>
    <div class="row">
      <div class="col-md-4">
        <div class="clearfix form-side-help">
          <div class="form-side-help-icon person">
          </div>
          <h2>Datos personales</h2>
        </div>
        <p>Esta información nos ayudará a verificar tu identidad.</p>
      </div>
      <div class="col-md-7 col-md-offset-1" style="padding-top: 20px;">
        <label for="">Nombre</label>
        <div class="row">
          <div class="col-sm-6">
            <div class="form-group ">
  <input class="form-control" id="id_first_name" maxlength="300" name="first_name" placeholder="Nombres" type="text" />
  <span class="help-block"></span>
</div>
          </div>
          <div class="col-sm-6">
            <div class="form-group ">
  <input class="form-control" id="id_last_name" maxlength="300" name="last_name" placeholder="Apellido(s)" type="text" />
  <span class="help-block"></span>
</div>
          </div>
        </div>

        <div class="row">
          <div class="col-sm-6">
            <div class="form-group ">
  <label for="id_identification">Número de cédula</label>
  <input class="form-control" id="id_identification" maxlength="10" name="identification" placeholder="0923335874" type="text" />
  <div class="help-block">
    
  </div>
</div>
          </div>
          <div class="col-sm-6">
            <div class="form-group ">
  <label for="id_position">Cargo</label>
  <input class="form-control" id="id_position" maxlength="150" name="position" placeholder="Contador" type="text" />
  <div class="help-block">
    
  </div>
</div>
          </div>
        </div>
      </div>
    </div>
    <hr>

    <div class="row">
      <div class="col-md-4 col-sm-3">
        <div class="clearfix form-side-help">
          <div class="form-side-help-icon shop">
          </div>
          <h2>Tu negocio</h2>
        </div>
        <p>Datos de tu empresa o de la empresa para la que trabajas.</p>
      </div>

      <div class="col-md-7 col-md-offset-1 col-sm-10" style="padding-top: 20px;">

        <div class="form-group ">
  <label for="id_legal_name">Razón social</label>
  <input class="form-control" id="id_legal_name" maxlength="300" name="legal_name" placeholder="ABC Compañía" type="text" />
  <div class="help-block">
    
  </div>
</div>
        <div class="form-group ">
  <label for="id_business_name">Nombre comercial</label>
  <input class="form-control" id="id_business_name" maxlength="300" name="business_name" placeholder="Café Gourmet" type="text" />
  <div class="help-block">
    
  </div>
</div>

        <div class="row">
          <div class="col-sm-6">
            <div class="form-group ">
  <label for="id_business_id">RUC</label>
  <input class="form-control" id="id_business_id" maxlength="13" name="business_id" placeholder="090901020001" type="text" />
  <div class="help-block">
    
  </div>
</div>
          </div>
        </div>
      </div>
    </div>

    <div class="accept-terms ">
      <div class="centered">
        <div class="checkbox">
          <label>
            <input class="" id="id_accept_terms" name="accept_terms" type="checkbox" /> He leído y acepto los Términos y Condiciones de Dátil
          </label>
        </div>
      </div>
    </div>

    <div class="form-buttons-fixed-bar">
      <div class="container">
        <div>
          <div class="centered">
            <button type="submit" id="submit"
                    class="btn btn-primary btn-lg"
                     disabled="" >
              Registrarme
            </button>
          </div>
        </div>
      </div>
    </div>

  </form>
```

This is the JS.

```js

  $(document).ready(function() {
    $("#id_email").attr(
      "data-parsley-remote",
      "/cuentas/email-disponible/VAL".replace('VAL', '{value}')
    );
    $("#id_username").attr(
      "data-parsley-remote",
      "/cuentas/usuario-disponible/VAL".replace('VAL', '{value}')
    );
    $("form").parsley(window.ParsleyConfig);
  });

```

**Comment by marcandre** [comment]
I don't understand, your submit button is disabled?
Do you have a live example?

**Comment by jplaza** [comment]
Parsley is working as expected

http://codepen.io/jplaza/pen/YyJYPb

It seems that some other JS is interfering with it. Any hint on where to look? I'm not getting any errors, it just won't submit.

This is my website: https://stagingapp.datil.co/cuentas/registro/personas

And this is my bower file:

``json
{
  "name": "datil-py",
  "version": "0.0.0",
  "dependencies": {
    "bootstrap": "~3.3",
    "bootstrap-select": "1.6.3",
    "font-awesome": "~4.2.0",
    "modernizr": "~2.6.2",
    "jquery": "~1.11",
    "react": "~0.13.3",
    "accounting.js": "~0.3.2",
    "query-string": "~1.0.0",
    "parsleyjs": "~2.2",
    "alt-checkbox": "~1.0.1",
    "dropzone": "~4.0.1"
  },
  "devDependencies": {}
}
```

**Comment by Alexx-G** [comment]
I've same issue with 2.1.3.
Upgrading to 2.2 solved my problem. But 2.2 is RC...
Can I use 2.2 version in production? Do you introduce any breaking changes?

**Comment by marcandre** [comment]
Yeah, it's a bit crazy to have an rc for so long, sorry about that. The change to ES6 took much longer than expected.
I also wanted to figure out a solution for validation interdependencies before freezing the new validation API, but I guess it's a bit late anyways.
My plan now is to release a last rc that fixes a small issue and a week later the actual 2.2.0.
API for `whenValid{ate}` will probably change in the next `rc` though, otherwise shouldn't be API change.

**Comment by marcandre** [comment]
Released `2.2.0`
